zebra/zebra_nb_state.c

```diff
@@ -392,37 +586,109 @@ const void *lib_vrf_ribs_rib_route_route_entry_nexthop_group_lookup_entry(
 
 /*
  * XPath:
- * /frr-vrf:lib/vrf/frr-zebra:ribs/rib/route/route-entry/nexthop-group/name
+ * /frr-vrf:lib/vrf/frr-zebra:zebra/ribs/rib/route/route-entry/nexthop-group/name
  */
 struct yang_data *
-lib_vrf_ribs_rib_route_route_entry_nexthop_group_name_get_elem(
+lib_vrf_zebra_ribs_rib_route_route_entry_nexthop_group_name_get_elem(
 	const char *xpath, const void *list_entry)
 {
-	/* TODO: implement me. */
+	struct nhg_hash_entry *nhe = (struct nhg_hash_entry *)list_entry;
+
+	if (nhe) {
+		char name[20] = {'\0'};
+
+		snprintf(name, sizeof(name), "%" PRIu32, nhe->id);
+		return yang_data_new_string(xpath, name);
+	}
 	return NULL;
 }
 
 /*
  * XPath:
- * /frr-vrf:lib/vrf/frr-zebra:ribs/rib/route/route-entry/nexthop-group/frr-nexthops/nexthop
+ * /frr-vrf:lib/vrf/frr-zebra:zebra/ribs/rib/route/route-entry/nexthop-group/frr-nexthops/nexthop
  */
 const void *
-lib_vrf_ribs_rib_route_route_entry_nexthop_group_frr_nexthops_nexthop_get_next(
+lib_vrf_zebra_ribs_rib_route_route_entry_nexthop_group_frr_nexthops_nexthop_get_next(
 	const void *parent_list_entry, const void *list_entry)
 {
-	/* TODO: implement me. */
-	return NULL;
+	struct nexthop *nexthop = (struct nexthop *)list_entry;
+	struct nhg_hash_entry *nhe = (struct nhg_hash_entry *)parent_list_entry;
+
+	if (list_entry == NULL)
+		nexthop = (nhe->nhg.nexthop);
+	else
+		nexthop = nexthop_next(nexthop);
+
+	return nexthop;
 }
 
-int lib_vrf_ribs_rib_route_route_entry_nexthop_group_frr_nexthops_nexthop_get_keys(
+int lib_vrf_zebra_ribs_rib_route_route_entry_nexthop_group_frr_nexthops_nexthop_get_keys(
 	const void *list_entry, struct yang_list_keys *keys)
 {
-	/* TODO: implement me. */
+	struct nexthop *nexthop = (struct nexthop *)list_entry;
+	char buf[PREFIX2STR_BUFFER];
+
+	keys->num = 3;
+
+	strlcpy(keys->key[0], yang_nexthop_type2str(nexthop->type),
+		sizeof(keys->key[0]));
+
+	switch (nexthop->type) {
+	case NEXTHOP_TYPE_IPV4:
+	case NEXTHOP_TYPE_IPV4_IFINDEX:
+		inet_ntop(AF_INET, &nexthop->gate.ipv4, buf, sizeof(buf));
+		strlcpy(keys->key[1], buf, sizeof(keys->key[1]));
```

Comment:
nit: you could use `snprintfrr(..., "%pI4", &nexthop->gate.ipv4);` instead to avoid the need for `buf` and `inet_ntop`.